docs/side_quests/ide_features.md

```diff
@@ -0,0 +1,654 @@
+
+Click on any issue to jump directly to the problematic line
+
+![Problems Panel](img/problems_panel.png)
+
+### 3.3. Common Error Patterns
+
+The extension helps identify:
+
+- **Missing brackets**: Unmatched `{` or `}`
+- **Incomplete blocks**: Missing required sections in processes
+- **Invalid syntax**: Malformed Nextflow DSL
+- **Typos in keywords**: Misspelled process directives
+- **Channel mismatches**: Type incompatibilities
+
+### Takeaway
+
+You can use VS Code's error detection and Problems panel to catch syntax errors and issues before running your workflow, saving time and preventing frustration.
+
+### What's next?
+
+Learn how to efficiently navigate between processes, modules, and definitions in complex workflows.
+
+---
+
+## 4. Code Navigation and Symbol Management
+
+Efficient navigation is crucial when working with complex workflows spanning multiple files. To understand this, replace the process definition in `basic_workflow.nf` with an import for the module we've provided you:
+
+```groovy
+include { FASTQC } from './modules/fastqc.nf'
+```
+
+### 4.1. Go to Definition
+
+Mouse over the `FASTQC` process name in the workflow block. You'll see a tooltip with the definition:
+
+![Go to definition](img/syntax.png)
+
+This feature is particularly valuable when authoring workflows, as it allows you to understand the module interface without opening the module file directly.
+
+You can quickly navigate to any process, module, or variable definition using **Ctrl-click** (or **Cmd-click** on Mac). Mouse over the link to the module file at the top of the script, and follow the link as suggested:
+
+![Follow link](img/follow_link.png)
+
+Try this on the `FASTQC` process name in the workflow block. This links you directly to the process name (which is the same as the module file in this example, but could be part-way through a much larger file).
+
+Now let's explore navigation in a more complex workflow using `complex_workflow.nf` (the illustration-only file mentioned earlier). This workflow contains multiple processes defined in separate module files, as well as some inline ones. While complex multi-file structures can be challenging to navigate manually, the ability to jump to definitions makes exploration much more manageable.
+
+1. Open `complex_workflow.nf`
+2. Ctrl-click on different process names
+3. Navigate to module definitions
+4. Use **Alt+←** (or **Ctrl+-**) to navigate back
+
+### 4.2. Symbol Navigation
+
+With `complex_workflow.nf` still open, you can get an overview of all symbols in the file by pressing **Ctrl+Shift+O** (or **Cmd+Shift+O**):
+
+![Symbol navigation](img/symbols.png)
+
+This shows:
+
+- All process definitions
+- Workflow definitions (there are two workflows defined in this file)
+- Function definitions
+
+Start typing to filter results.
+
+### 4.3. Workspace Symbol Search
+
+Press **Ctrl+T** (or **Cmd+T**) to search symbols across your entire workspace:
```

Comment:
```suggestion
Press `Ctrl+T` (or `Cmd+T`) to search symbols across your entire workspace:
```

this one doesn't work at all. I couldn't find an alternative command (the desc says it's correct), but cmd+t opens a new browser tab in chrome and takes precedence